src/pytorch_lightning/trainer/trainer.py

```diff
@@ -963,6 +988,18 @@ def _restore_modules_and_callbacks(self, checkpoint_path: Optional[_PATH] = None
     def _run(
         self, model: "pl.LightningModule", ckpt_path: Optional[str] = None
     ) -> Optional[Union[_EVALUATE_OUTPUT, _PREDICT_OUTPUT]]:
+        if model._compiler_ctx is not None:
```

Comment:
I think it would be cleaner for the strategy to implement a validation mechanism directly and for DDPStrategy / DDPFullyShardedNativeStrategy to uncompile the model and wrap their own forward model after wrapping.

Reply:
Agreed, this will be the way forward in the future, as soon as we understand more about dynamo and distributed.

Reply:
@lantiga I agree with @tchaton. This could be implemented as a [plugin](https://github.com/Lightning-AI/lightning/tree/master/src/pytorch_lightning/plugins) that is specifically enabled for some of the strategies, Which would contain all this patching and unpatching logic.

If you agree, I would like to give it a shot